output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/io/ChangedNoticesCollector.java

```diff
@@ -0,0 +1,121 @@
+package org.mobilitydata.gtfsvalidator.outputcomparator.io;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.mobilitydata.gtfsvalidator.model.ValidationReport;
+import org.mobilitydata.gtfsvalidator.outputcomparator.model.report.AffectedSource;
+import org.mobilitydata.gtfsvalidator.outputcomparator.model.report.ChangedNotice;
+
+/**
+ * Given validation reports computed against two versions of the validator across multiple datasets,
+ * collects the set of newly-generated error notices that changed between the two validator
+ * versions.
+ */
+public class ChangedNoticesCollector {
+
+  /**
+   * If the number of new error notices in a report meets or exceeds this threshold, the
+   * corresponding dataset will be considered "invalid".
+   */
+  private final int newErrorThreshold;
```

Comment:
I'm wondering if this shouldn't be a percentage threshold as well, instead of being an absolute value. 

In practice, 1 new error in a dataset that already has over 100 errors is not much, but 10 new errors in a dataset that previously had 2 errors seems like a lot. @isabelle-dr do you have thoughts on this?

Reply:
See my comment above 😊. I'd like this test to capture as much information as possible.